pyspeckit/spectrum/classes.py

```diff
@@ -193,6 +193,8 @@ def __init__(self, filename=None, filetype=None, xarr=None, data=None,
 
         if unit is not None:
             self._unit = unit
+        # else:
+        #     self._unit = u.dimensionless_unscaled
```

Comment:
I left this in just so you can have a look at it, it addresses the issue https://github.com/pyspeckit/pyspeckit/issues/48 where Cube._unit was not set.
I'm not sure if this the way to do it because when I uncommented it, somewhere in the writer there was an error: CompositeUnit is not iterable when it searched for the "$" (LaTeX related if I remember correctly)